.pre-commit-config.yaml

```diff
@@ -0,0 +1,43 @@
+ci:
+  autoupdate_commit_msg: "chore: update pre-commit hooks"
+  autofix_commit_msg: "style: pre-commit fixes"
+
+repos:
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: "v0.0.292"
+    hooks:
+      - id: ruff
+        args: [--fix, --ignore=E741, --exclude=__init__.py]
+
+  - repo: https://github.com/nbQA-dev/nbQA
+    rev: 1.7.0
+    hooks:
+      - id: nbqa-ruff
+        additional_dependencies: [ruff==0.0.284]
+        args: ["--fix","--ignore=E501,E402"]
+
+  # - repo: https://github.com/adamchainz/blacken-docs
```

Comment:
do we need this or can be deleted?

Reply:
Agreed, I thought of leaving it for when we have the docs up and sorted, but we can just add it then.